Information/Design.md

```diff
@@ -89,3 +89,320 @@
+
+
+**REQ 150**
+
+
+_The bot shall allow privileged users edit data to the database_
+
+
+
+**REQ 160**
+
+_The bot shall allow users to track the monsters they have killed_
+
+
+**REQ 170**
+
+_The bot shall allow users to track the monsters that have killed them_
+
+
+**REQ 180**
+
+_The bot shall be able to display a list of all available commands supported by the bot_
+ 
+
+------------------------------------------------------------------------------------------
+
+
+######REQ 130 Design: 
+
+
+Command name: _-item make_
+
+Parameters: _name_, _type_, _rarity_ 
+
+**Design of _item make_ command**:
+
+The name for the command, _item_ ,was a straight forward process. I needed a name that would 
+non ambiguous.  In contrast, if the command name was _-i_ or _-itm_. There leaves a level of 
+decoding from the users perspective, on trying to figure out the function of the command. 
+Where as _-item make_ does not require a significant amount more typing. Yet leaves no ambiguity
+as to the function of the command. Moreover, the second word in the command, _make_, 
+is easier to spell for people than say _craft_ or _recipe_. 
+
+The parameters of the _item make_ command are a way to provide a convenient way to search for a 
+item recipe. The user need not know exactly what he/she wishes to search for. However,
+this command does require that the user needs to know exactly one of the three ways of 
+quantifying an item in the Monster Hunter World game. Providing no parameters, returns 
+nothing, except a message telling the user to input a name, type, or rarity if they wish
+to see a craft-able recipe for that item, if applicable. Not all items in the Monster 
+Hunter World game are craft-able. Thus if the item searched for is non-craft-able then the
+bot will return a message to that affect. If the user inputs two or all three parameters 
+the _item make command_ shall use the first parameter given. This design provides 
+flexibility and robustness from an functionality standpoint. Moreover, makes it a convenient
+user friendly design so that the user does not need to be an expert in all things related
+to craft-able items in Monster Hunter World. In order to find out useful information about 
+items and craft-ability.  
+
+The flexibility of the parameters was the most intellectually intensive part of this command's
+design. From the users perspective, if they don't know anything about the game or very little
+I wanted to be able to still have this Bot provide useful information. Thus entering one, two
+or all three parameters will not only cause no functionality undefined errors. Moreover, 
+will provide a flex-able input interface for the user. Thus need know only one of the three
+ways to represent an item that is craft-able. 
+
+
+examples of _item make_ command usage: 
+
+`-item make Mega Potion`  -- example with _name_ parameter 
+
+`-item make medicine`     -- example with _type_ parameter
```

Comment:
In the implementation process, you should not compare either or of name, or type parameter.  IF the user supplies only one or the other, then the command will still work. If the user provides both of them.  Then it can either use one or the other, or user both, in case one produces a result and one does not.  Moreover, if the user miss spells one or the other than the command can still produce an output of desired results.